src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtils.java

```diff
@@ -0,0 +1,1914 @@
+            logger.warn("createAminoAcidSequence given a coding sequence of length not divisible by 3.  Dropping bases from the end: " + (codingSequence.length() % 3));
+        }
+
+        for ( int i = 0; i < maxIndex; i += 3 ) {
+            final AminoAcid aa = getEukaryoticAminoAcidByCodon(codingSequence.substring(i, i+3));
+            if ( aa == null ) {
+                throw new UserException.MalformedFile("File contains a bad codon sequence that has no amino acid equivalent: " + codingSequence.substring(i, i+3));
+            }
+            else {
+                sb.append(aa.getLetter());
+            }
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Get the coding sequence-aligned allele based on stop and start position.
+     * @param codingSequence Coding sequence from which the allele should be derived.  Must not be {@code null}.
+     * @param alignedAlleleStart Start position of the allele (1-indexed, inclusive).  Must not be {@code null}.  Must be > 0.
+     * @param alignedAlleleStop Stop position of the allele (1-indexed, inclusive).  Must not be {@code null}.  Must be > 0.
+     * @param strand {@link Strand} on which the allele is coded.  Must not be {@code null}.  Must not be {@link Strand#NONE}.
+     * @return The {@link String} representation of the allele.
+     */
+    private static String getAlignedAlleleSequence(final String codingSequence,
+                                                   final Integer alignedAlleleStart,
+                                                   final Integer alignedAlleleStop,
+                                                   final Strand strand) {
+        Utils.nonNull(codingSequence);
+        Utils.nonNull(alignedAlleleStart);
+        ParamUtils.isPositive( alignedAlleleStart, "Genome positions must be > 0." );
+        Utils.nonNull(alignedAlleleStop);
+        ParamUtils.isPositive( alignedAlleleStop, "Genome positions must be > 0." );
+        assertValidStrand( strand );
+
+        // Get our indices:
+        // Subtract 1 because we're 1-based.
+        int start = alignedAlleleStart - 1;
+        int end = alignedAlleleStop;
+
+        final String alignedAlleleSeq;
+
+        if ( strand == Strand.POSITIVE ) {
+            alignedAlleleSeq = codingSequence.substring(start, end);
+        }
+        else {
+            // Negative strand means we need to reverse complement and go from the other end:
+            start = codingSequence.length() - alignedAlleleStop;
+            end = codingSequence.length() - alignedAlleleStart + 1;
+            alignedAlleleSeq = ReadUtils.getBasesReverseComplement( codingSequence.substring(start, end).getBytes() );
+        }
+
+        return alignedAlleleSeq;
+    }
+
+    /**
+     * Gets the coding sequence for the allele with given start and stop positions, codon-aligned to the start of the reference sequence.
+     * @param codingSequence The whole coding seque
```

Comment:
Perhaps this class should be in a separate `SequenceComparison.java` source file, instead of embedded in this utils class.

Reply:
Fixed!